system-addon/test/functional/mochitest/browser_highlights_section.js

```diff
@@ -0,0 +1,69 @@
+"use strict";
+
+registerCleanupFunction(async () => {
+  await clearHistoryAndBookmarks();
+});
+
+function refreshHighlightsFeed() {
+  // Toggling the pref will clear the feed cache and force a places query.
+  Services.prefs.setBoolPref("browser.newtabpage.activity-stream.feeds.section.highlights", false);
+  Services.prefs.setBoolPref("browser.newtabpage.activity-stream.feeds.section.highlights", true);
+}
+
+async function add_highlights_bookmark(count) {
+  const bookmarks = new Array(count).fill(null).map((entry, i) => ({
+    parentGuid: PlacesUtils.bookmarks.unfiledGuid,
+    title: "foo",
+    url: `https://mozilla${i}.com/nowNew`
+  }));
+
+  for (let placeInfo of bookmarks) {
+    await PlacesUtils.bookmarks.insert(placeInfo);
+  }
+
+  // Add bookmark visits
+  for (let placeInfo of bookmarks) {
+    await PlacesTestUtils.addVisits(placeInfo.url);
+  }
+
+  // Force HighlightsFeed to make a request for the new items.
+  refreshHighlightsFeed();
+}
+
+async function addHighlightsCards(browser, count) {
+  await clearHistoryAndBookmarks();
+  await add_highlights_bookmark(count);
+  await ContentTask.spawn(browser, null, async () => {
+    await ContentTaskUtils.waitForCondition(() => content.document.querySelector(".card-outer:not(.placeholder)"),
+      "No highlights cards found.");
```

Comment:
A description of what we're expecting to change would be useful. And the failure message isn't entirely right I think? Can we end up showing empty state instead of 3 placeholders? E.g., `test_highlights(0, test_empty_state)`